pkg/server/credtemplate/builder.go

```diff
@@ -223,9 +224,11 @@ func (b *Builder) BuildDownstreamX509CATemplate(ctx context.Context, params Down
 	tmpl.Subject = params.ParentChain[0].Subject
 	tmpl.Subject.OrganizationalUnit = []string{fmt.Sprintf("DOWNSTREAM-%d", len(params.ParentChain))}
 
-	// It's a bit gross, but SPIRE has historically signed downstream X509CA's with the X509-SVID ttl, so
-	// let's override the NotBefore/NotAfter fields set by buildX509CATemplate.
-	tmpl.NotBefore, tmpl.NotAfter = b.computeX509SVIDLifetime(params.ParentChain, params.TTL)
+	if b.config.UseLegacyDownstreamX509CATTL {
+		// It's a bit gross, but SPIRE has historically signed downstream X509CA's with the X509-SVID ttl, so
+		// let's override the NotBefore/NotAfter fields set by buildX509CATemplate.
+		tmpl.NotBefore, tmpl.NotAfter = b.computeX509SVIDLifetime(params.ParentChain, params.TTL)
```

Comment:
this used to change default value when ttl == 0, but now that it is no longer called, what happens in that scenario?

Reply:
The call to buildX509CATemplate above applies the default X509 CA TTL. This call to computeX509SVIDLifetime was the hack to apply the SVID default instead.